lib/ukrust/ukrust_sys/lib.rs

```diff
@@ -60,3 +60,8 @@ fn panic(_info: &core::panic::PanicInfo<'_>) -> ! {
         __ukrust_sys_crash();
     }
 }
+
+#[lang = "eh_personality"]
+#[no_mangle]
+pub extern fn rust_eh_personality() {
+}
```

Comment:
This should not be necessary as well. This requires changes to the target and might or might not be in scope of this PR.

Reply:
Removing this affects compiling `app-helloworld-rust`:

```
/usr/bin/ld: /home/fabi/my_unikernel/apps/app-helloworld-rust/build/cargo/./libapphelloworld.a(core-05898138a596088a.core.178bbcc6-cgu.0.rcgu.o):(.data.DW.ref.rust_eh_personality[DW.ref.rust_eh_personality]+0x0): undefined reference to `rust_eh_personality'
collect2: error: ld returned 1 exit status
```

Do you know more about this behaviour?

Reply:
I think this is caused by the target we're currently using. As Martin said, this should be tackled in a future PR.